File: Lv.2/행렬의 곱셈.cpp
#include <string>
#include <vector>

using namespace std;
// 1 4  3 3
// 3 2  3 3
// 4 1
//          j
// i 2 3 2  5 4 3
//   4 2 4  2 4 1
//   3 1 4  3 1 1
// 
// 1   3  4
// 2
vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer(arr1.size(),vector<int>(arr2[0].size(),0));
    
    for(int i=0;i<arr1.size();i++){
        for(int j=0;j<arr2[0].size();j++){
            for(int k=0;k<arr1[i].size();k++){
                answer[i][j] += arr1[i][k] * arr2[k][j];
            }
        }
    }
    return answer;
}

File: Lv.2/행렬의 곱셈.js
function solution(arr1, arr2) {
    const cross = [];

    for (let i = 0 ; i < arr1.length ; i ++) {
        const rows = [];
        for(let j = 0 ; j < arr2[0].length ; j ++) {
            rows.push(0);
        }
        cross.push(rows);
    }

    for (let i = 0 ; i < arr1.length ; i ++) {
        for (let j = 0 ; j < arr1[0].length ; j ++) {
            for (let k = 0 ; k < arr2[0].length ; k ++) {
                cross[i][k] += arr1[i][j] * arr2[j][k];
            }
        }
    }

    return cross;
}